doc/user/content/transform-data/strategies-considerations.md

```diff
@@ -0,0 +1,159 @@
+menu:
+  main:
+    parent: "transform-data"
+    weight: 95
+    name: "Strategies and considerations"
+---
+
+Materialize provides various [indexing strategies to optimize query
+performance](/transform-data/optimization/). However, because indexes in
+Materialize maintain the complete query results in memory (not just the index
+keys and the pointers to data rows), ... [will be updated].
+
+The following lists some strategies and considerations that may help you better
+utilize the benefits provided by Materialize while managing
+[costs](/administration/billing/).
+
+## Strategies to reduce dataset/memory usage
+
+### Projections
+
+Indexes store the complete query results in memory, not just the index keys and
+the pointers to data rows. As such, avoid extraneous fields in your query
+projections. That is, if you are interested in a smaller
+subset of the fields in your dataset, select only those fields instead of using
+`SELECT *`.
+
+### Temporal filters
+
+If you are only interested in the data from a recent period of time (e.g., the
+last 7 days, the last 24 hours, etc.), you can use [temporal
+filters](/transform-data/patterns/temporal-filters) to create a [sliding
+window](/transform-data/patterns/temporal-filters/#sliding-window) over the
+dataset.
+
+Considerations:
+
+- Determine if you need to add a grace period in order to account for late
+  arriving data (e.g., records that arrive out of order due to network issues).
+
+- If using stacked views with temporal filters, avoid unnecessarily duplicating
+  the same temporal filter up the stack. Since Materialize maintains all future
+  retractions of the data that is currently within the temporal window (in
+  anticipation of when data will fall outside the window), removing unnecessary
+  temporal filters help reduce memory use.
+
+- [*Private preview*](/releases/previews/). Set an expiration for maintaining
+  future retractions. When using temporal filters, Materialize (in anticipation
+  of when data will fall outside the window) maintains all future retractions of
+  the data that is currently within the temporal window. Depending on your
+  dataset size and window size, you may want to configure Materialize to drop
+  expired retractions.
+
+### Static datasets
+
+Specifying [`REFRESH AT
+CREATION`](/sql/create-materialized-view/#refresh-strategies) for your
+materialized view effectively flags that view as static.
+
+If your data model consists of large datasets that join with smaller, static
+(i.e., does not change) datasets (such as a dimensional data model where your
+dimensions do not change), consider using materialized views with a [`REFRESH AT
+CREATION`](/sql/create-materialized-view/#refresh-strategies) specification for
+your static datasets. Joins with a static dataset may be able to reduce the
+amount of memory required for the other join inputs.
+
+### Hot/warm/cold data patterns
+
+If the freshness requirements of your workload vary such that some staleness is
```

Comment:
Depending on the specific details, implementing this pattern can become very complicated very quickly. Does it make sense to highlight when it works reasonably well to better guide customers when to use it? Right now it reads as if it just solves all problems, which it does not.

Reply:
Ah, definitely would be good to highlight when to use it as well as gotchas.